samtranslator/model/events.py

```diff
@@ -24,28 +22,13 @@ class EventsRule(Resource):
 
     runtime_attrs = {"rule_id": lambda self: ref(self.logical_id), "arn": lambda self: fnGetAtt(self.logical_id, "Arn")}
 
-    def __init__(
-        self,
-        logical_id: Optional[Any],
-        relative_id: Optional[str] = None,
-        depends_on: Optional[List[str]] = None,
-        attributes: Optional[Dict[str, Any]] = None,
-    ) -> None:
-        super().__init__(logical_id, relative_id, depends_on, attributes)
-
-        if len(self.logical_id) > EVENT_RULE_LOGICAL_ID_MAX_LENGTH:
-            # Truncate logical id to satisfy the EVENT_RULE_ID_MAX_LENGTH limit
-            self.logical_id = _truncate_with_suffix(
-                self.logical_id, EVENT_RULE_LOGICAL_ID_MAX_LENGTH, EVENT_RULE_LOGICAL_ID_EVENT_SUFFIX
-            )
-
 
 def generate_valid_target_id(logical_id: str, suffix: str) -> str:
```

Comment:
I feel 2 functions here is a little verbose. Maybe combining into one function?

Reply:
Oops. My bad... Was looking into how to bypass the checks. Didn't realize you have another comment. Will fix this next time I have a PR.